test/e2e/memcached_test.go

```diff
@@ -569,8 +577,88 @@ func MemcachedClusterTest(t *testing.T) {
 	cmdOut, err := exec.Command("operator-sdk", "test", "cluster", *e2eImageName,
 		"--namespace", namespace,
 		"--image-pull-policy", "Never",
-		"--service-account", "memcached-operator").CombinedOutput()
+		"--service-account", operatorName).CombinedOutput()
 	if err != nil {
 		t.Fatalf("In-cluster test failed: %v\nCommand Output:\n%s", err, string(cmdOut))
 	}
 }
+
+func memcachedMetricsTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Make sure metrics Service exists
+	s := v1.Service{}
+	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: operatorName, Namespace: namespace}, &s)
+	if err != nil {
+		return fmt.Errorf("could not get metrics Service: (%v)", err)
+	}
+
+	// Get operator pod
+	pods := v1.PodList{}
+	opts := client.ListOptions{Namespace: namespace}
+	if err := opts.SetLabelSelector(fmt.Sprintf("name=%s", operatorName)); err != nil {
+		return fmt.Errorf("failed to set list label selector: (%v)", err)
+	}
+	if err := opts.SetFieldSelector("status.phase=Running"); err != nil {
+		return fmt.Errorf("failed to set list field selector: (%v)", err)
+	}
+	err = f.Client.List(context.TODO(), &opts, &pods)
+	if err != nil {
+		return fmt.Errorf("failed to get pods: (%v)", err)
+	}
+	podName := ""
+	if len(pods.Items) > 1 {
+		// If we got more than one pod, get leader pod name.
+		leader, err := verifyLeader(t, namespace, f)
+		if err != nil {
+			return err
+		}
+		podName = leader.Name
+	}
```

Comment:
It doesn't look like we test that the label selectors we use on the `Service` resolve to the operator pods we care about. Instead of doing the `PodList` stuff, we could get the `Endpoints` object associated with the `Service`, and iterate the `Endpoint`'s `subset.addresses` list looking for the leader pod name.

If the leader pod name doesn't exist, that would be another failure condition. And if it does, we pick right back up at line 626.

WDYT?

Reply:
> It doesn't look like we test that the label selectors we use on the `Service` resolve to the operator pods we care about.

Yes we do, we use the `Deployment` label in this case to get all the pods that are selected by the `Service` to get the list of `Pod`s.

Reply:
> If the leader pod name doesn't exist, that would be another failure condition. And if it does, we pick right back up at line 626.

True! Added a check for that as well. PTALA.

Reply:
> Yes we do, we use the `Deployment` label in this case to get all the pods that are selected by the `Service` to get the list of `Pod`s.

I guess what I mean is that we hardcode the label selector in line 602, which matches our `Service` selector now, but may not always. Since we test that the `Service` exists, I was suggesting that we go one step further to either use its label selector or its associated `Endpoints`.

If we do that, this test is future-proofed to changes we may make to the label selectors on the `Service` and labels on the `Deployment` pod spec.